### PR TITLE
Support service registration on branch snapshots published to metadata

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary
Support service registration on branch snapshots published to metadata

## How did you test this change?

- [x] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-graph/src/graph-manager/action/service/ServiceExecutionServer.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export enum ServiceExecutionServer {
+  INT = 'INT',
```
Comment (gayathrir11) [inline packages/legend-graph/src/graph-manager/action/service/ServiceExecutionServer.ts:18]: you don't need this enum. You can get this info from the config `TEMPORARY__serviceRegistrationConfig`

File: packages/legend-application-studio/src/stores/project-view/ProjectViewerStore.ts [hunk 2]
```diff
@@ -181,6 +184,7 @@ export class ProjectViewerStore {
 
     // fetch project versions
     yield flowResult(this.editorStore.sdlcState.fetchProjectVersions());
```
Comment (gayathrir11) [inline packages/legend-application-studio/src/stores/project-view/ProjectViewerStore.ts:186]: Can you check with Mauricio if we still need the sdlc flow to fetch the project versions. The only place we are using this now is ProjectOverviewState. If we can just use depot flow to get the versions there?

Comment (MauricioUyaguari) [reply packages/legend-application-studio/src/stores/project-view/ProjectViewerStore.ts:186]: I think for now we can leave it for the project overview state. we can think later if we want to change it.

File: packages/legend-graph/src/graph-manager/action/service/ServiceExecutionServer.ts (hunk 1 above)
Comment (MauricioUyaguari) [reply packages/legend-graph/src/graph-manager/action/service/ServiceExecutionServer.ts:18]: yes agree we shouldnt need to fetch it.

File: packages/legend-application-studio/src/stores/editor/EditorDepotState.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  assertErrorThrown,
+  type PlainObject,
+} from '@finos/legend-shared';
+import { StoreProjectData } from '@finos/legend-server-depot';
+import { LEGEND_STUDIO_APP_EVENT } from '../../__lib__/LegendStudioEvent.js';
+
+export class EditorDepotState {
+  readonly editorStore: EditorStore;
+
+  currentProject?: StoreProjectData | undefined;
```
[the remark this answers is not in the document]
Comment (MauricioUyaguari) [reply packages/legend-application-studio/src/stores/editor/EditorDepotState.ts:32]: yeah why are we fetching current project. we should have groupId,artifactID present

[the remark this answers is not in the document]
Comment (YannanGao-gs) [reply packages/legend-application-studio/src/stores/editor/EditorDepotState.ts:32]: tried to use `      this.projectVersions =
        (yield this.editorStore.depotServerClient.getVersions(
          this.editorStore.projectConfigurationEditorState
            .currentProjectConfiguration.groupId,
          this.editorStore.projectConfigurationEditorState
            .currentProjectConfiguration.artifactId,
          true,
        )) as string[];`
somehow currentProjectConfiguration is undefined.. 👎

File: packages/legend-graph/src/graph-manager/action/service/ServiceExecutionServer.ts (hunk 1 above)
Comment (YannanGao-gs) [reply packages/legend-graph/src/graph-manager/action/service/ServiceExecutionServer.ts]: removed! use a const instead

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**